lib/dns/resource.js

```diff
@@ -311,17 +313,24 @@ class Resource extends Struct {
     // Potentially an answer.
     const res = new Message();
 
+    // TLDs are authoritative over their own NS & TXT records.
+    // The NS records in the root zone are just "hints"
+    // and therefore are not signed by the root ZSK.
+    // The only records root is authoritative over is DS.
     switch (type) {
       case types.NS:
         res.authority = this.toNS(name);
         res.additional = this.toGlue(name);
-        key.signZSK(res.authority, types.NS);
         break;
       case types.TXT:
-        res.answer = this.toTXT(name);
-        key.signZSK(res.answer, types.TXT);
+        if (!this.hasNS()) {
+          res.aa = true;
+          res.answer = this.toTXT(name);
+          key.signZSK(res.answer, types.TXT);
```

Comment:
Should the PR description be updated to say we _do_ sign `TXT`? And `AA` is set for `TXT` as well?

> Root doesn't sign TXT or NS with ZSK, only DS
> Root doesn't set authority bit for TXT or NS, only DS

Reply:
thank you sir! updated the PR description to reflect final changes after #565 discussion